### PR TITLE
Fix missing feature in starky

### PR DESCRIPTION
**Pull request by eigmax**
- [x] support AVX512 acceleration for starky.  https://github.com/0xEigenLabs/eigen-zkvm/commit/6f52173775bc3f5f8046d56411eee4cd36e81686 .  The feature `stdarch_x86_avx512` is not available in Rust's stable channel.

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "1.77"
+channel = "nightly"
```
Comment (Schaeff) [inline rust-toolchain.toml:2]: Is this change required?

Comment (eigmax) [reply rust-toolchain.toml:2]: Yes, this is required by AVX512.  And that's why I try to merge it into powdr-labs:avx

Comment (eigmax) [reply rust-toolchain.toml:2]: There is another options:  creating a new rust-toolchain in backend,  I am not sure which one is better.

**Review by leonardoalt — APPROVED** [review]

**Comment by leonardoalt** [comment]
@eigmax the tests are still failing due to `feature` being used with a stable channel. For the purposes of this branch we can also make the whole repo `nightly` so you can run some experiments.

**Comment by leonardoalt** [comment]
@eigmax ok I'll just merge this PR then regardless of CI
